### PR TITLE
run, create, connect: add support for gw-priority

### PR DESCRIPTION
**Pull request by akerouanton**
Related to:

- https://github.com/moby/moby/pull/48936

**- What I did**

This adds support for the newly introduced `GwPriority` API field. It can be set on both `docker container run` and `docker network connect`.

For the `run` command, it's made available only through the extended syntax.

**- How I did it**

**- How to verify it**

CI is green.

**- Description for the changelog**

```markdown changelog
- Added a new `gw-priority` option to `docker run`, `docker container create`, and `docker network connect`. This option will be used by the Engine to determine which network provides the default gateway for a container. On `docker run`, this option is only available through the extended `--network` syntax.
```

### CODE REVIEW

File: cli/command/network/connect.go [hunk 1]
```diff
@@ -23,6 +23,7 @@ type connectOptions struct {
 	aliases      []string
 	linklocalips []string
 	driverOpts   []string
+	gwPriority   int
```
Comment (thaJeztah) [inline cli/command/network/connect.go:26]: I'm probably a bit late realising, but should this have been a uint, or do we expect negative values to be ok? 🙈

Comment (akerouanton) [reply cli/command/network/connect.go:26]: It's an int on purpose as it allows to set the priority to a negative value when some existing endpoints have the default priority `0` set. @robmry wrote about that here: https://github.com/moby/moby/pull/48936#discussion_r1858809270

Comment (thaJeztah) [reply cli/command/network/connect.go:26]: But wasn't higher higher priority, so if I want a new network to become the default, I need to give it a higher priority?

Comment (akerouanton) [reply cli/command/network/connect.go:26]: Absolutely. But if one of the current network give you a default gateway, and you want to stick with it, you can connect new networks with a negative priority.

Comment (thaJeztah) [reply cli/command/network/connect.go:26]: (mostly considering if it's zero, then I did not consider it a priority (and alphabetic order is used), and only if I would explicitly want a network to take priority, I'd probably set a non-zero value; any value lower (including zero) would not take priority in that  case)

**Review by robmry — APPROVED** [review]

File: cli/command/network/connect.go (hunk 1 above)
Comment (akerouanton) [reply cli/command/network/connect.go:26]: You might consider priority of existing endpoints only when you connect a new network. In that case, having the ability to set a negative priority gives you a way to not touch the existing endpoint(s), but still get the behavior you want.

**Review by thaJeztah — APPROVED** [review]
LGTM

**Pull request merged**